src/Data/VectorSpace/Core.agda

```diff
@@ -0,0 +1,94 @@
+------------------------------------------------------------------------
+-- The Agda standard library
+--
+-- Abstract vector spaces.
+--
+-- A "vector space" is a Module with a commutative, homomorphic inner
+-- product and a complete set of building blocks for mapping the space.
+------------------------------------------------------------------------
+
+{-# OPTIONS --without-K --safe #-}
+
+module Data.VectorSpace.Core where
+
+import Algebra.Module.Morphism.Structures as MorphismStructures
+
+open import Algebra        using (CommutativeRing)
+open import Algebra.Module using (Module)
+open import Algebra.Module.Construct.TensorUnit using (⟨module⟩)
+open import Algebra.Module.Morphism.Bundles
+open import Data.List      using (List; foldr)
+open import Data.Product
+open import Function
+open import Level          using (Level; _⊔_; suc)
+open import Relation.Binary
+open import Relation.Binary.ExtensionalEquivalence
+open import Relation.Binary.Reasoning.MultiSetoid
+
+private
+  variable
+    a b c : Level
+    A : Set a
+    B : Set b
+    C : Set c
+
+record VectorSpace
+  {r ℓr m ℓm : Level}
+  {ring      : CommutativeRing r ℓr}
+  (mod       : Module ring m ℓm)
+  : Set (suc (ℓr ⊔ r ⊔ ℓm ⊔ m)) where
+
+  constructor mkVS
```

Comment:
I refuse to believe that this constructor with 15+ arguments is actually useful :laughing: